compiler/src/iree/compiler/Codegen/Common/test/reconcile_translation_info.mlir

```diff
@@ -543,3 +543,95 @@ hal.executable private @multi_export_scf_forall {
+//  CHECK-DAG:   %[[C1:.+]] = arith.constant 1
+//  CHECK-DAG:   %[[WG_X1:.+]] = arith.constant 64
+//  CHECK-DAG:   %[[WG_X:.+]] = arith.maxui %[[WG_X0]], %[[WG_X1]]
+//      CHECK:   hal.return %[[WG_X]], %[[C1]], %[[C1]]
+//      CHECK: func @multiple_scf_forall_ops()
+//  CHECK-DAG:   %[[C32:.+]] = arith.constant 32
+//  CHECK-DAG:   %[[C16:.+]] = arith.constant 16
+//  CHECK-DAG:   %[[UB0:.+]] = hal.interface.constant.load {{.+}} ordinal(0)
+//  CHECK-DAG:   %[[UB1:.+]] = hal.interface.constant.load {{.+}} ordinal(1)
+//  CHECK-DAG:   %[[STEP0:.+]] = hal.interface.constant.load {{.+}} ordinal(2)
+//  CHECK-DAG:   %[[STEP1:.+]] = hal.interface.constant.load {{.+}} ordinal(3)
+//  CHECK-DAG:   %[[EXTENT0:.+]] = affine.apply #[[MAP1]]()[%[[UB0]], %[[STEP0]]]
+//  CHECK-DAG:   %[[EXTENT1:.+]] = affine.apply #[[MAP1]]()[%[[UB1]], %[[STEP1]]]
+//  CHECK-DAG:   %[[WG_ID_X:.+]] = hal.interface.workgroup.id[0]
+//  CHECK-DAG:   %[[WG_COUNT_X:.+]] = hal.interface.workgroup.count[0]
+
+//  CHECK-DAG:   %[[WG_IDS_1ST:.+]]:2 = affine.delinearize_index %[[WG_ID_X]]
+// CHECK-SAME:     into (%[[EXTENT0]], %[[EXTENT1]])
+//  CHECK-DAG:   %[[NPROCS1_1ST:.+]] = affine.min #[[MAP2]]()[%[[UB1]], %[[STEP1]], %[[WG_COUNT_X]]]
+//  CHECK-DAG:   %[[NPROCS0_1ST:.+]] = affine.min #[[MAP3]]()[%[[UB0]], %[[STEP0]], %[[WG_COUNT_X]], %[[NPROCS1_1ST]]]
+//  CHECK-DAG:   %[[LOOP_START0_1ST:.+]] = affine.apply #[[MAP4]]()[%[[WG_IDS_1ST]]#0, %[[STEP0]]]
+//  CHECK-DAG:   %[[LOOP_START1_1ST:.+]] = affine.apply #[[MAP4]]()[%[[WG_IDS_1ST]]#1, %[[STEP1]]]
+//  CHECK-DAG:   %[[LOOP_STEP0_1ST:.+]] = affine.apply #[[MAP4]]()[%[[NPROCS0_1ST]], %[[STEP0]]]
+//  CHECK-DAG:   %[[LOOP_STEP1_1ST:.+]] = affine.apply #[[MAP4]]()[%[[NPROCS1_1ST]], %[[STEP1]]]
+//  CHECK-NOT:   scf.forall
+//      CHECK:   scf.for %[[I:.+]] = %[[LOOP_START0_1ST]] to %[[UB0]] step %[[LOOP_STEP0_1ST]]
+//      CHECK:     scf.for %[[J:.+]] = %[[LOOP_START1_1ST]] to %[[UB1]] step %[[LOOP_STEP1_1ST]]
+//      CHECK:       "use"(%[[I]], %[[J]])
+// CHECK-NEXT:     }
+// CHECK-NEXT:   }
+
+//  CHECK-DAG:   %[[WG_IDS_2ND:.+]]:2 = affine.delinearize_index %[[WG_ID_X]] into (16, 4)
+//  CHECK-DAG:   %[[NPROCS1_2ND:.+]] = affine.min #[[MAP5]]()[%[[WG_COUNT_X]]]
+//  CHECK-DAG:   %[[NPROCS0_2ND:.+]] = affine.min #[[MAP6]]()[%[[WG_COUNT_X]], %[[NPROCS1_2ND]]]
+//  CHECK-DAG:   %[[LOOP_START1_2ND:.+]] = affine.apply #[[MAP8]]()[%[[WG_IDS_2ND]]#0]
+//  CHECK-DAG:   %[[LOOP_START0_2ND:.+]] = affine.apply #[[MAP7]]()[%[[WG_IDS_2ND]]#1]
+//  CHECK-DAG:   %[[LOOP_STEP1_2ND:.+]] = affine.apply #[[MAP8]]()[%[[NPROCS0_2ND]]]
+//  CHECK-DAG:   %[[LOOP_STEP0_2ND:.+]] = affine.apply #[[MAP7]]()[%[[NPROCS1_2ND]]]
+//      CHECK:   scf.for %[[I:.+]] = %[[LOOP_START0_2ND]] to %[[C16]] step %[[LOOP_STEP0_2ND]]
+//      CHECK:     scf.for %[[J:.+]] = %[[LOOP_START1_2ND]] to %[[C32]] step %[[LOOP_STEP1_2ND]]
+//      CHECK:       "use"(%[[I]], %[[J]])
+// CHECK-NEXT:     }
+// CHECK-NEXT:   }
```

Comment:
A test with different `mapping_id` sizes (e.g. one with all 3, xyz, and another with just x and y) would be good. Don't need as high fidelity verification of the IR, just loop nest sizes + workgroup count region.